src/Mjolnir.ts

```diff
@@ -803,9 +807,10 @@ export class Mjolnir {
 
         // Check for updated ban lists before checking protected rooms - the ban lists might be protected
         // themselves.
-        if (this.banLists.map(b => b.roomId).includes(roomId)) {
+        const banList = this.banLists.find(list => list.roomId === roomId);
+        if (banList !== undefined) {
             if (ALL_RULE_TYPES.includes(event['type'])) {
-                await this.syncWithPolicyRoom(roomId);
+                banList.updateForEvent(event)
```

Comment:
No it still does a fetch of the room state exactly like we did before (even though it is unnecessary), we just manage it from `BanList` instead so that the batching mechanism can be contained inside `BanList` (where it does belong, instead of polluting Mjolnir or ApplyBan/ACL).   

Redactions were never handled via sync and it's not been updated to handle them either. If rules have been redacted then they will only be removed when the BanList is updated because of another event. (Because the redacted rules will be shown in the room state that BanList.update asks for)

We probably do want to use this as the opportunity to change it to update when we get a redaction though, so i will change that. FTR the way Mjolnir removes rules itself is not with a redaction, but by replacing the rule with an empty one.

Reply:
> We probably do want to use this as the opportunity to change it to update when we get a redaction though, so i will change that. FTR the way Mjolnir removes rules itself is not with a redaction, but by replacing the rule with an empty one.

Is that still planned? If not, this can definitely wait for a followup.

Reply:
Oh, it's already been changed